### PR TITLE
Removing chunks on failed to write to replicas

### PR DESCRIPTION
**Pull request by kmlebedev**
https://github.com/seaweedfs/seaweedfs/issues/3578

# What problem are we solving?

https://github.com/seaweedfs/seaweedfs/issues/3578

# How are we solving the problem?

https://github.com/seaweedfs/seaweedfs/issues/3578

# How is the PR tested?

# Checks
- [ ] I have added unit tests if possible.
- [ ] I will add related wiki document changes and link to this PR after merging.

### CODE REVIEW

File: weed/server/filer_server_handlers_write_upload.go [hunk 1]
```diff
@@ -197,6 +198,15 @@ func (fs *FilerServer) dataToChunk(fileName, contentType string, data []byte, ch
 	})
 	if err != nil {
 		glog.Errorf("upload error: %v", err)
+		if strings.Contains(err.Error(), "failed to write to replicas for volume") {
```
Comment (chrislusf) [inline weed/server/filer_server_handlers_write_upload.go]: there could be multiple retries. Need to put this within the util.Retry section.

**Comment by kmlebedev** [comment]
@chrislusf Hello, I also wanted to clarify what happens when a chunk is re-written locally to a volume?
Will there be a rewrite or no change?

**Comment by chrislusf** [comment]
Both are possible.

**Comment by kmlebedev** [comment]
@chrislusf fixed

File: weed/server/filer_server_handlers_write_upload.go [hunk 2]
```diff
@@ -192,28 +195,27 @@ func (fs *FilerServer) dataToChunk(fileName, contentType string, data []byte, ch
 		if uploadErr != nil {
 			glog.V(4).Infof("retry later due to upload error: %v", uploadErr)
 			stats.FilerRequestCounter.WithLabelValues(stats.ChunkDoUploadRetry).Inc()
+			if strings.Contains(uploadErr.Error(), "failed to write to replicas for volume") {
```
Comment (chrislusf) [inline weed/server/filer_server_handlers_write_upload.go]: Depending on string is not reliable.

How about just purge on any errors?

Comment (kmlebedev) [reply weed/server/filer_server_handlers_write_upload.go]: > How about just purge on any errors?

@chrislusf Let's try. Done

**Pull request merged**